app/payment-link-v2/reference-confirm/reference-confirm.njk

```diff
@@ -1,13 +1,49 @@
 {% extends "../../views/layout-payment-links-v2.njk" %}
-{% from "govuk/components/button/macro.njk" import govukButton %}
-{% from "govuk/components/input/macro.njk" import govukInput %}
 {% from "govuk/components/back-link/macro.njk" import govukBackLink %}
-{% from "../../views/macros/error-summary.njk" import errorSummary %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
 
 {% block pageTitle %}
-  {{ __p('paymentLinksV2.referenceConfirm.areYouSureThisIsAValid') }} {{product.reference_label}}  - {{ productName }}
+  {{ isThisAValidReferenceText }}  - {{ productName }}
 {% endblock %}
 
 {% block contentBody %}
-  <p class="govuk-body">Reference Confirm page - placeholder text</p>
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <h1 class="govuk-heading-l">{{ productName }}</h1>  
+
+    {% if productDescription %}
+      <p class="govuk-body">{{ productDescription }}</p>
+    {% endif %}
+
+    <h2 class="govuk-heading-m">{{ paymentReferenceLabel }}</h2>
+
+    {% if paymentReferenceHint %}
+      <p class="govuk-hint">{{ paymentReferenceHint }}</p>    
+    {% endif %}
+
+    {{ govukInsetText({
+      text: "4242424242424242",
+      classes: "govuk-!-font-weight-bold"
+    }) }}
+
+
+    <h2 class="govuk-heading-m">{{ isThisAValidReferenceText }}</h2>
+
+    <p class="govuk-body">{{ __p('paymentLinksV2.referenceConfirm.checkYouveEnteredTheNumberCorrectly') }}</p>
+
+    <div class="govuk-button-group">
+      {{ govukButton({
+        text: __p('paymentLinksV2.referenceConfirm.confirmAndContinue'),
+        href: confirmPageUrl
+      }) }}
+
+      {{ govukButton({
+        text: __p('paymentLinksV2.referenceConfirm.edit'),
+        classes: "govuk-button--secondary",
+        href: editPageUrl
+      }) }}
```

Comment:
Do these buttons need to be swapped round? I think the appearance should be unchanged from the existing page.

Reply:
Good spot - forgot to mention that in the PR.\
Seems like in the Design system, they usually have the Primary button at the start - so I thought it may be better that ways.  

Please let me know what you think?

<img width="949" alt="image" src="https://user-images.githubusercontent.com/59831992/156615490-5e05387a-733d-4f01-b0ce-015d1d2d725b.png">

Reply:
Ah cool, sounds good. Happy to approve this, do you want to run it by Johnny at some point?

I think we need to look at the overall content and design of this page in particular anyway as part of the review as we're currently displaying the payment link title and description when on other pages we've removed these for the new design.  This page wasn't looked at in the firebreak discovery.